Bio/SeqRecord.py

```diff
@@ -159,15 +162,15 @@ class SeqRecord:
 
     def __init__(
         self,
-        seq,
+        seq: Union["Seq", "MutableSeq"],
         id="<unknown id>",
         name="<unknown name>",
         description="<unknown description>",
         dbxrefs=None,
         features=None,
         annotations=None,
         letter_annotations=None,
-    ):
+    ) -> None:
```

Comment:
May I ask how you decided which function argument should be type-hinted in this PR? Just curious, since in this function, for example, it's only seq that's type-hinted while the others are left untouched. Not asking that they should all be type-hinted by the way, just why some are and the others are not :).

Reply:
Probably because the ``SeqRecord`` initialisation was this only bit being used directly in the modules being type hinted? But I'm curious too.

Reply:
My process
1. `mypy Bio/SeqIO/QualityIO.py`
2. `mypyc Bio/SeqIO/QualityIO.py`  (since this was the goal, and it throws up some slightly different error messages)
3. `mypy --strict Bio/SeqIO/QualityIO.py | grep QualityIO` to catch functions missing types and to find type errors elsewhere that may be of interest

Reply:
Any of the above commands might highlight classes and functions in other files that also would benefit (or in mypyc's case, require) to get type hints